File: app/src/main/java/org/autojs/autojs/ui/widget/BindableViewHolder.java
package org.autojs.autojs.ui.widget;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

/**
 * Created by Stardust on Apr 8, 2017.
 */
public abstract class BindableViewHolder<DataType> extends RecyclerView.ViewHolder{


    public BindableViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bind(DataType data, int position);
}

File: app/src/main/java/org/autojs/autojs/ui/widget/BindableViewHolder.kt
package org.autojs.autojs.ui.widget

import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by Stardust on Apr 8, 2017.
 * Transformed by SuperMonster003 on Nov 23, 2024.
 */
abstract class BindableViewHolder<DataType>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    abstract fun bind(data: DataType, position: Int)

    open fun onViewRecycled() {
        /* Nothing to do by default. */
    }

}
